### PR TITLE
Consolidate

### PR DESCRIPTION
**Pull request by andrew103**
Part of this PR also includes changing the name of this repo to `trajectory_executor` and deleting the [trajectory_tracker](https://github.com/WPI-MMR/trajectory_tracker) repo

### CODE REVIEW

File: package.xml [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
 <?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
 <package format="3">
-  <name>trajectory_generator</name>
+  <name>trajectory_executor</name>
   <version>0.0.0</version>
-  <description>ROS node for generating a new trajectory based on current pose and desired end pose</description>
+  <description>ROS package that handles all aspects required in executing a trajectory</description>
   <license>MIT</license>
-  <url type="website">https://github.com/WPI-MMR/trajectory_generator</url>
+  <url type="website">https://github.com/WPI-MMR/trajectory_executor</url>
```
Comment (mahajanrevant) [inline package.xml:8]: TODO: Remember to actually change the repository name

File: trajectory_executor/tracker.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    timer_period = 2 # seconds
+    self.timer = self.create_timer(timer_period, self.timer_callback)
+    self.ard_check = self.create_timer(0.1, self.ard_check_callback)
+
+    self.arduino_port = serial.Serial(
+      port="/dev/ttyAMA1",  # TODO: evaluate if we should change to cli flag
+      baudrate=115200
+    )
+
+  # TODO: Add docstring and type hints
```
Comment (mahajanrevant) [inline trajectory_executor/tracker.py]: Can we do this within this pull request? It's going to keep getting pushed back

File: trajectory_executor/tracker.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
+import rclpy
+from rclpy.node import Node
+
+from trajectory_interfaces.msg import Trajectory, SensorData
+
+import serial
+
+class TrajectoryTracker(Node):
+
```
Comment (mahajanrevant) [inline trajectory_executor/tracker.py:8]: nit: extra spaces. Remove them

File: trajectory_executor/tracker.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
+import rclpy
+from rclpy.node import Node
+
+from trajectory_interfaces.msg import Trajectory, SensorData
+
+import serial
+
```
Comment (mahajanrevant) [inline trajectory_executor/tracker.py:7]: Should be two spaces after import statements

File: trajectory_executor/tracker.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  def timer_callback(self):
+    msg = SensorData()
+    msg.data = 'Arduino Sensor Data'
+    self._publisher.publish(msg)
+
+  def new_traj_callback(self, msg):
+    self.get_logger().info(msg.data)
+
+
```
Comment (mahajanrevant) [inline trajectory_executor/tracker.py:48]: nit: Remove one space

File: trajectory_executor/tracker.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+def main(args=None):
```
Comment (mahajanrevant) [inline trajectory_executor/tracker.py:49]: Why even create this main method? Why not have this instead
```
if __name__ == '__main__':
  rclpy.init(args=args)
  trajectory_tracker = TrajectoryTracker()
  rclpy.spin(trajectory_tracker)
  rclpy.shutdown()
```

**Review by mahajanrevant — COMMENTED** [review]
Let me know if you have any questions?

File: trajectory_executor/tracker.py (hunk 3 above)
Comment (goobta) [reply trajectory_executor/tracker.py:8]: look at this guy using nits xD

File: trajectory_executor/tracker.py (hunk 6 above)
Comment (andrew103) [reply trajectory_executor/tracker.py:49]: callback function is a requirement for ROS2 to be able to find the package

File: trajectory_executor/tracker.py [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+  rclpy.init(args=args)
+
```
Comment (goobta) [inline trajectory_executor/tracker.py]: nit: do we need all of these spaces? gets confusing w/ python

**Review by goobta — APPROVED** [review]
revant basically got everything. otherwise lgtm

**Review by goobta — APPROVED** [review]
lgtm

**Review by mahajanrevant — APPROVED** [review]
Looks good!

**Pull request merged**